### ISSUE TITLE
Nav component: Header

### ISSUE THREAD

**Issue by fcoveram** [WordPress/wporg-main-2022#263]

**Comment by ryelle** [comment]
Whoops, I read too quickly and saw the global nav difference in the News screenshots and thought this was about the global nav— you had the right repo first 😅

**Comment by ryelle** [comment]
tl;dr — Idea 1 feels more straightforward, I have more questions about Idea 2.

> Idea 1 → Treat the header as a site menu that shows all L2 pages. For multiple L2 pages, the menu shifts to a dropdown, like in News. 

The idea mocked up in the Developers screenshot would be significantly easier to implement, since that could be a menu dropdown that uses the existing Nav block. For example, I could make that happen almost immediately:

![Screen Shot 2023-05-04 at 11 25 22](https://user-images.githubusercontent.com/541093/236255074-1c6f099f-a360-410a-af69-731417ac8264.png)

The News-style mega-menu dropdown ([like this](https://user-images.githubusercontent.com/541093/235984116-b260c125-dfe3-40a3-8fcb-43fc8fc2a3cf.png)) would need to be a custom block, and would probably mean we need to code up at least two variations of the local nav bar — one with a plain nav menu and one with the mega-menu. Again, this isn't meant to say we can't, just that it increases the complexity.

> In cases where the site is deeper, showing a breadcrumb might be necessary, but the main point is reinforcing the idea of section identities and how diverse the WordPress site is.

In the mockups with breadcrumbs, they're now outside of the local nav bar. Can we assume that the left part of the nav bar will always be the section title (ex, About, Download - or site title, when they're separate sites like Plugins, Themes)?

Will this be too many nav-ish bars when we're on archive pages of the directories? On the homepages they'll have the big header to separate nav & filter, but archives typically use the internal pages pattern, for example, Patterns

![patterns](https://user-images.githubusercontent.com/541093/236267241-256b809f-5236-40c6-a2d5-253944f105b4.png)

Also, are we dropping local search boxes from the header area?

semi-related but nitpicky, when this is finalized can we also set one breadcrumb separator for all instances? We've got [mid-dots](https://wordpress.org/news/category/meta/), [slashes](https://wordpress.org/documentation/article/debugging-in-wordpress/), and now arrows.

> Idea 2 → Show a breadcrumb that mixes a dropdown behavior to strengthen the site location. This work well where you dive into posts and jump between sections, like in “Documentation” and “Forums”.

Maybe I'm stuck with the "old site" way of doing things, but this seems like an unexpected interaction pattern to me. I feel like I wouldn't know what level of nav to appear under these dropdowns (sibling or child pages).

Would we stop using the nested breadcrumbs like [Documentation](https://wordpress.org/documentation/article/administer-your-blog/), or would it be possible to have multiple dropdowns?

It looks like this is still using the dropdown top right nav in some examples, would that be a simple nav dropdown or might we still need the mega-menu pattern in this version?

**Comment by jasmussen** [comment]
Do we need a border below the 2nd navbar?

**Comment by fcoveram** [comment]
> The idea mocked up in the Developers screenshot would be significantly easier to implement,

I don't follow the idea. The screenshot you added is based on **Idea 1**, but it shows the breadcrumb in the subnav area and the menu doesn't list all L2 pages, just the ones in "The people" category.

> In the mockups with breadcrumbs, they're now outside of the local nav bar. Can we assume that the left part of the nav bar will always be the section title (ex, About, Download - or site title, when they're separate sites like Plugins, Themes)?

I hear your point, and I'm not sure how to solve the name duplication. Having the page name in the subnav area and in the breadcrumb at the same time looks visually odd, but necessary for location and consistency. Here is a quick mockup of my point.

<img width="1163" alt="CleanShot 2023-05-05 at 12 34 30@2x" src="https://user-images.githubusercontent.com/895819/236436052-03440d23-4ea5-4c94-b316-7cf3541ccf86.png">

I'm open to any ideas for this.

> Will this be too many nav-ish bars when we're on archive pages of the directories?

In this transitional stage, it does look busy and stacked. But in the current redesigns, there is always a section between header and filter bar as the latter refreshes the content area placed below. This is visible in Showcase, Themes, and Patterns figma files.

> Also, are we dropping local search boxes from the header area?

This is open to discussion. But I'm drawn to move it to the hero section and make it consistent across Showcase, Themes, Patterns, and Plugins.

> when this is finalized can we also set one breadcrumb separator for all instances? We've got [mid-dots](https://wordpress.org/news/category/meta/), [slashes](https://wordpress.org/documentation/article/debugging-in-wordpress/), and now arrows.

Very much agree.

> …this seems like an unexpected interaction pattern to me. I feel like I wouldn't know what level of nav to appear under these dropdowns (sibling or child pages). Would we stop using the nested breadcrumbs like [Documentation](https://wordpress.org/documentation/article/administer-your-blog/), or would it be possible to have multiple dropdowns?

I was thinking of multiple dropdowns. The navigation is definitely not common and seeing sibling pages felt obvious to me as the breadcrumb layout shows parent pages on the left and child pages on the right. But I might be wrong so I would appreciate what other folks think.

> This is still using the dropdown top right nav in some examples, would that be a simple nav dropdown or might we still need the mega-menu pattern in this version?

For related links a simple dropdown. I can't envision the mega menu in other sections, it fits with News content only.

**Comment by ryelle** [comment]
> I don't follow the idea. The screenshot you added is based on Idea 1, but it shows the breadcrumb in the subnav area and the menu doesn't list all L2 pages, just the ones in "The people" category.

Ah, I spent about 5 min hacking this into an existing branch so yeah, it didn't cover everything. I just wanted to differentiate the default dropdown from the mega-menu design.

How's this attempt?

![](https://user-images.githubusercontent.com/541093/236505859-fa6a3a45-78c2-4524-8ab0-0ea39720faed.png)

> I hear your point, and I'm not sure how to solve the name duplication. Having the page name in the subnav area and in the breadcrumb at the same time looks visually odd, but necessary for location and consistency. Here is a quick mockup of my point.

I didn't really mean that as a critique, just confirming my assumption of what each item is in the mockup.

**Comment by StevenDufresne** [comment]
> I hear your point, and I'm not sure how to solve the name duplication. Having the page name in the subnav area and in the breadcrumb at the same time looks visually odd, but necessary for location and consistency. Here is a quick mockup of my point.

> [Original Image](https://github.com/WordPress/wporg-main-2022/issues/263#issuecomment-1536078606)

> I'm open to any ideas for this.

<img width="400" alt="236436052-03440d23-4ea5-4c94-b316-7cf3541ccf86" src="https://user-images.githubusercontent.com/1657336/236715144-b04434c8-f6fd-4fa9-b451-1193e6f6b797.png">

Is there an argument for using the top level category to tie the sub-sites together? I don't really like having breadcrumbs share a container that also has variable width. [See context](https://github.com/WordPress/wporg-mu-plugins/pull/348#issuecomment-1455554325).

**Comment by fcoveram** [comment]
To @ryelle's point

> How's this attempt?

The dropdown looks good, but splitting the L2 pages into three dropdowns would be confusing. [Idea 1](https://user-images.githubusercontent.com/895819/236237488-ebc0e719-0fb9-4f61-af4d-9c7ce65055e0.png) shows all pages in the available or one dropdown with all pages. For this case, it could look like the following mockup.

<img width="1243" alt="CleanShot 2023-05-08 at 14 27 21@2x" src="https://user-images.githubusercontent.com/895819/236823653-7d2bb227-d698-4e17-b2ac-9a90ec5073e0.png">

**Note:** The category sections and active style in the popover are for the argument and don't intend to establish a frontend decision.

To @StevenDufresne's point

> Is there an argument for using the top level category to tie the sub-sites together? I don't really like having breadcrumbs share a container that also has variable width. [See context](https://github.com/WordPress/wporg-mu-plugins/pull/348#issuecomment-1455554325).

I agree with not putting the breadcrumb next to another variable-width element. That's why Idea 1 places it below the subnavigation while Idea 2 has a single dropdown on the right side that occasionally exists. To your concern, Idea 1 fits better for tackling long breadcrumbs.

> I see one downside; the related links are based on sub-site and not top level category.

With the top level category you mean "Download & Extend", "Learn", and so on? If so, I don't understand why showing a related link dropdown in subsites (Themes, Documentation, for picking some) that nowadays have it in the subnav area could be a problem.

**Comment by ryelle** [comment]
The issue with a long dropdown is what happens on short screens, since the nav bar is sticky:

https://user-images.githubusercontent.com/541093/236874289-ad158436-5bab-499a-958c-7cebfa4a790c.mp4

I don't find it confusing that there are 3 dropdowns, to me it makes it clear the structure of the section. Besides, "the technology" "the details" etc are not real pages, which works nicely if these are  "click to open" nav menus.

Is the label for the dropdown meant to be the current page title? For consistency (and a11y), it would be better if the menu label was the same for all pages in a section (or said differently, any time a menu is used, it should have the same top-level label).

**Comment by fcoveram** [comment]
You're right. The dropdown does look long.

I explained myself wrongly. It's not confusing in terms of understanding their purpose and what displays, but confusing as it adds a new subnav variant, either if we go with idea 1 or 2. This would be the only site section with three dropdowns and I think we should avoid creating customized navigations.

In that vein, the [hiding the menu](https://github.com/WordPress/wporg-main-2022/issues/255#issuecomment-1530866007) suggestion looks better to me as the About WordPress section has a component at the end of the page listing all pages.

> Is the label for the dropdown meant to be the current page title?

That was the initial idea. But if it affects a11y, then we would need to find a static label. "Menu" is the first idea that comes to me, but it feels outdated.

**Comment by ryelle** [comment]
>  It's not confusing in terms of understanding their purpose and what displays, but confusing as it adds a new subnav variant, either if we go with idea 1 or 2.

I don't see how this is a significantly different variation on Idea 1 — this is how menus work. Technically, there's no difference between a flat menu, a single submenu, & multiple submenus, it's all just a Navigation block.

There's also already a possible second dropdown in the header with "related links" (and I don't think that would exist on About), so it's just one more jump to the 3-submenu list.

> In that vein, the https://github.com/WordPress/wporg-main-2022/issues/255#issuecomment-1530866007 suggestion looks better to me as the About WordPress section has a component at the end of the page listing all pages.

I'm fine with that solution too — I'm only pushing about this to make sure we have something that will work in all future use cases too.

> That was the initial idea. But if it affects a11y, then we would need to find a static label. "Menu" is the first idea that comes to me, but it feels outdated.

Maybe it's dated, but that just means it's understandable, IMO. Though that could be another point in the 3-dropdown menu's favor, we can use the section titles as top-level menu items (won't help much for other sites, though).

**Comment by StevenDufresne** [comment]
This is a great discussion, let me try to summarize so we can align and find action items.

**Points**:
1. We appear to be leaning towards Idea 1.
1. We are leaning towards having breadcrumbs under the subnav and not in it.
1. We aren't sure whether to use the Sub-site title or the top level category title in the subnav, above the breadcrumbs
1. We like the idea of a dropdown for subsites with many L2 pages, but listing them vertically would make the menu too long. We could use the news mega menu, just a bit more work.
1.  Instead of a mega menu, we could go with multiple dropdown menus. We're a bit hesitant because we would have both a mega menu paradigm and a multi menu paradigm. We should probably choose only one.
1. For the dropdown, we aren't sure what label to add when the dropdown is closed, probably just "menu" or "pages".
1. We are tempted to remove the menu items from the about section.
1. We are leaning towards removing the search bar from the nav and repurposing it elsewhere in the pages/directories.

Is this a fair assessment of where we are at in this discussion?

**Comment by fcoveram** [comment]
Going back here after several days.

Now that Showcase designs are done, I want to share and idea I had when made the first wireframes on how global and subsite headers can work together.

https://github.com/WordPress/wporg-main-2022/assets/895819/01bece7e-385a-4e39-8eab-685bbd32385a

The idea's core is to keep the subsite navigation visible to provide more context of the content displayed as subsites are different to each. Specially when landing on a specific page without going through the global nav.

Both headers have small spacing tweaks to make them look similar, and global nav items that expand a menu in hover are now single elements that show the menu with a click. This latter point is based on the idea of disabling the redirect to the first page, like `Download & Extend > Get WordPress` and `Community > Make WordPress`.

I'm unaware of its feasability, but sharing it here to sparkle discussion and see your thoughts (cc @WordPress/meta-design)

**Comment by jasmussen** [comment]
I like it, it feels valid to me, and should mostly be a matter of applying a `position: sticky;`

**Comment by ryelle** [comment]
`position: sticky` would only control the stickiness, not the hide/appear logo & local site name, or the menu appearing on scroll up. We can give building it a try if this is a go, but just to be clear, it's definitely not going to be that simple 🙂 

Personally, I dislike the menu-on-scroll-up since it changes the header size, often it ends up covering the thing i'm scrolling up to read.

**Comment by StevenDufresne** [comment]
I tend to agree with @ryelle. These menus feel nice but I don't find the links compelling. I don't know why I would be browsing the showcase sites and decide I all-of-a-sudden want to submit a site and even "view all sites" because that doesn't provide me with a more useful view of sites until I exhaust the list i'm currently viewing (get to the bottom). I would also make the same argument for most of our sub-sites.

**Comment by jasmussen** [comment]
This may be more valid on some sections, like Documentation, Learn, Developer Resources, than Showcase. I don't know _if_ we'd want this as a generic one for all.

**Comment by ryelle** [comment]
For what it's worth, my issue isn't with the local site navigation, it's the fact that the global nav appears when you scroll up (see at 0:21 on the video above).

> I don't know if we'd want this as a generic one for all.

So if a site like showcase didn't have a local header, would the global header still be sticky at the top of the page when scrolling (current behavior)? If so, there's a divergence, and we should be very clear about when that happens.

**Comment by jasmussen** [comment]
Right, that the "home" logo fades in! I'd be happy to omit that bit entirely,so it's literally only the local header that sticks, and no change to the content of it.

And no, I don't think we'd want any divergence if we could avoid it. The sticky is most useful for the local header.

**Comment by fcoveram** [comment]
I agree with the odd effect caused by two headers.

The main reason for showing it was to keep the global navigation visible "all the time," acknowledging the scroll-up interaction to see it.

But I'm very drawn to only show the local navigation ✨

**Comment by ryelle** [comment]
This scrolling question has come up again since Showcase now needs breadcrumbs, so I'd like to get this sorted out. For easier reference, I've mocked up the different options in code & have some screen recordings. The admin bar is visible for me because I have access to the site. For most people it would not be visible.

First, a quick reference for what these are called in code & block names so we're all discussing the same things.

![names](https://github.com/WordPress/wporg-main-2022/assets/541093/bae0c82e-d41f-4556-bdcd-2df38c5fabe0)

## Sticky variations

Global header stays sticky, local nav bar is sticky, breadcrumbs are not sticky. This is basically the current behavior if we just drop the breadcrumbs in.

https://github.com/WordPress/wporg-main-2022/assets/541093/ff50d9b5-4066-4446-aa07-54d53bfb6cca

---

Everything is sticky — this takes over ~250px of screen height, so I doubt we want this :)

https://github.com/WordPress/wporg-main-2022/assets/541093/b848ac05-3d20-423d-a358-9410f6e866f0

---

Global header is not sticky, local nav bar is sticky, and breadcrumbs are sticky. This changes the global header behavior on all sites, not just ones with local navigations.

https://github.com/WordPress/wporg-main-2022/assets/541093/f1f566e4-32ec-42b6-b20b-c710e0b8656f

The unstickied global header on a classic theme

https://github.com/WordPress/wporg-main-2022/assets/541093/7292c729-f983-4f21-8902-ae241f2ae97e

---

Global header is not sticky, local nav bar is sticky, and breadcrumbs are not sticky. This version only has one sticky bar, but it might be weird that it's the middle bar?

https://github.com/WordPress/wporg-main-2022/assets/541093/aa35957e-87d6-4e81-b0d2-15c8c12d9bd9

---

I have the code for all of this in local branches, and I'd love to figure it out before I go on sabbatical, just so it's in place for all the other redesign sites (this already affects [Documentation](https://wordpress.org/documentation/category/where-to-start/) & [Developer](https://developer.wordpress.org/redesign-test/cli/commands/post/) which took different directions 😑 ). 

cc @WordPress/meta-design @ndiego @adamwoodnz @StevenDufresne @renintw

**Comment by jasmussen** [comment]
I would lean towards two-bars.mp4, with one-bar.mp4 as another option that would be acceptable. 

I believe both of those also honor [Franciscos concept here](https://github.com/WordPress/wporg-main-2022/issues/263#issuecomment-1665356186), even if we won't be making the logo-moving-down change.

**Comment by marko-srb** [comment]
Yep, @fcoveram's proposal seems great, for scroll down (loose global nav) when start going back (appear global nav)...

**Comment by ryelle** [comment]
Right, this expands on @fcoveram's idea, trying to flesh out how it should work in all cases, not just the ideal redesign homepage case, and iterating based on the conversation that followed.

So for "two bars": It would show both local nav & breadcrumbs when both exist, or just local nav when there are no breadcrumbs, and no sticky header at all on classic themes.

"One bar" would leave only the local nav sticky, regardless of breadcrumbs.

> putting back global nav whenever user scrolls up...

I feel strongly about **not** doing that, it's probably my most hated web pattern TBH. A header appearing like that almost always covers the thing I've just scrolled up to read, then I end up wiggling my scroll to get rid of it, and now I'm annoyed at the website. I would rather keep the 3 bars sticky always and reduce vertical space than surprise people with randomly-appearing banners. 

> even if we won't be making the logo-moving-down change.

I did add the "logo moving down" in my prototype, you can see it in both two-bars.mp4 and one-bar.mp4.

**Comment by ryelle** [comment]
Yes, the prototype was working code :)

Should I make that into a PR? It might be a good idea to bundle this change in with the other header menu changes, since it will change the behavior of the global header menu.

**Comment by jasmussen** [comment]
I'm into it! I also know you've got some AFK coming up, might be nice to store in a PR for someone else to potentially pick up.
